### PR TITLE
Update packages, Fix Bugs and add format method

### LINKED ISSUE #1: Perform transform in-memory, not on-disk
Requires https://github.com/palantir/tslint/issues/1760

### PR DESCRIPTION
**Pull request by orblazer**
Hello,
This PR contains :
* Update all packages
* Fix the import on CLI
* Use relatative package (prettier and tslint) when run that
* Fix mock FS
* Add format method for prettier-vscode (this commit contains "applyTSLintAllFixes" for fix tslint errors in text and NOT in file)
This partial fix #1

This PR is required for : https://github.com/prettier/prettier-vscode/pull/414

Thanks.

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -0,0 +1,295 @@
+import requireRelative from "require-relative";
+/* import merge from "lodash.merge";
```
Comment (azz) [inline src/utils.js]: Can this big block be deleted?

File: src/format.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+ *   Will default to require.resovlve('prettier')
+ * @param {Object} options.tslintConfig - the config to use for formatting
+ *  with TSLint.
+ * @param {Object} options.prettierOptions - the options to pass for
+ *  formatting with `prettier`. If not provided, prettier-eslint will attempt
+ *  to create the options based on the eslintConfig
+ * @param {Object} options.fallbackPrettierOptions - the options to pass for
+ *  formatting with `prettier` if the given option is not inferrable from the
+ *  eslintConfig.
+ * @param {String} options.logLevel - the level for the logs
```
Comment (azz) [inline src/format.js]: can delete this option

File: src/apply-fixes.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export function applyTSLintAllFixes(linterResults, source, tslint) {
```
Comment (azz) [inline src/apply-fixes.js]: This is a lot simpler than I imagined this being...

File: package.json [hunk 4]
```diff
@@ -11,26 +11,27 @@
...
+    "ignore": "^3.3.7",
+    "require-relative": "^0.8.7",
+    "tslint": "^5.9.1",
+    "yargs": "^11.0.0"
   },
   "peerDependencies": {
-    "prettier": "^1.7.4",
-    "typescript": "^2.5.3"
+    "prettier": "^1.12.0",
+    "typescript": "^2.8.1"
```
Comment (azz) [inline package.json]: Is there a reason you increased the `peerDependencies`? I don't think we need any of the newer APIs?

File: src/utils.js (hunk 1 above)
Comment (orblazer) [reply src/utils.js]: The comment yes

File: src/format.js (hunk 2 above)
Comment (orblazer) [reply src/format.js]: OK

File: src/apply-fixes.js (hunk 3 above)
Comment (orblazer) [reply src/apply-fixes.js]: Yes, me too im suprised by simplicity :)

File: package.json (hunk 4 above)
Comment (orblazer) [reply package.json]: no is not required but its good to have last versions

Comment (azz) [reply package.json]: This is `peerDependencies`, which forces anyone who uses `prettier-tslint` to upgrade to latest `typescript` and `prettier`, which we don't want to do.

**Comment by azz** [comment]
Thanks heaps! Release should be automatically published soon.

**Comment by azz** [comment]
https://github.com/azz/prettier-tslint/releases/tag/v0.4.0

**Pull request merged**